addons/mail/static/src/discuss/message_pin/common/thread_model_patch.js

```diff
@@ -0,0 +1,50 @@
+import { patch } from "@web/core/utils/patch";
+import { Record } from "@mail/core/common/record";
+import { Thread } from "@mail/core/common/thread_model";
+
+import { rpcWithEnv } from "@mail/utils/common/misc";
+let rpc;
+
+patch(Thread.prototype, {
+    setup() {
+        super.setup();
+        rpc = rpcWithEnv(this.env);
```

Comment:
I'm wondering if it'd less annoying to have a `mail.rpc` service instead... What do you think?

This approach was used to keep the same look as the `rpc` function of `@web/core/network/rpc_service`... But it's also cumbersome to juggle with `rpcWithEnv` and to bind on `env` by hand.

I don't have a strong preference, both require shenanigans in discuss code... But probably one is slightly worse than the other

Reply:
I don't know what is the issue in the first place. I think import rpc should just work.

Reply:
RPC can be triggered at end of a test, which can mess up following tests.
To solve this issue, a discuss test are aware of env(s) made in the test and accepts only RPCs with such env(s). At end of test, these envs are removed from eligible envs, so that promises are infinitely pending.

Note that at the time I wasn't aware that js odoo modules / files were scoped for each test by HOOT system, so there's probably a way to use these scopes without having to pollute the business code. The mock server is shared among tests but not most JS files, so probably we could bind the `env` "globally" on the (scoped) `rpc`.
Not entirely sure this would work, but if it does we can have normal `rpc()` without `rpcWithEnv()` function in business code.